src/ProductTour/Checkpoint.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState } from 'react';
+import { useMediaQuery } from 'react-responsive';
+import PropTypes from 'prop-types';
+import { createPopper } from '@popperjs/core';
+
+import CheckpointActionRow from './CheckpointActionRow';
+import CheckpointBody from './CheckpointBody';
+import CheckpointBreadcrumbs from './CheckpointBreadcrumbs';
+import CheckpointTitle from './CheckpointTitle';
+
+function Checkpoint({
+  body,
+  index,
+  placement,
+  target,
+  title,
+  totalCheckpoints,
+  ...props
+}) {
+  const [checkpointVisible, setCheckpointVisible] = useState(false);
+  const isMobile = useMediaQuery({ query: '(max-width: 768px)' });
+
+  useEffect(() => {
+    const targetElement = document.querySelector(target);
+    const checkpoint = document.querySelector('#checkpoint');
+    if (targetElement && checkpoint) {
+      // Use the Popper library to translate the Checkpoint to its target's coordinates
+      const checkpointPopper = createPopper(targetElement, checkpoint, {
+        placement: isMobile ? 'top' : placement,
+        modifiers: [
+          {
+            name: 'arrow',
+            options: {
+              padding: 25,
+            },
+          },
+          {
+            name: 'offset',
+            options: {
+              offset: [0, 20],
+            },
+          },
+          {
+            name: 'preventOverflow',
+            options: {
+              padding: 20,
+              tetherOffset: 35,
+            },
+          },
+        ],
+      });
+      setCheckpointVisible(true);
+      if (checkpointPopper) {
+        checkpointPopper.forceUpdate();
+      }
+    }
+  }, [target, isMobile]);
+
+  useEffect(() => {
+    if (checkpointVisible) {
+      // Scroll the Checkpoint into view once its rendered
+      const targetElement = document.querySelector(target);
+      let targetOffset = targetElement.getBoundingClientRect().top;
+      if ((targetOffset < 0) || (targetElement.getBoundingClientRect().bottom > window.innerHeight)) {
+        if (placement.includes('top')) {
+          if (targetOffset < 0) {
+            targetOffset *= -1;
+          }
+          targetOffset -= 280;
+        } else {
+          targetOffset -= 80;
+        }
+
+        window.scrollTo({
+          top: targetOffset, behavior: 'smooth',
+        });
+      }
+
+      const button = document.querySelector('#pgn__checkpoint-primary-button');
+      button.focus();
+    }
+  }, [target, checkpointVisible]);
+
+  const isLastCheckpoint = index + 1 === totalCheckpoints;
+  const isOnlyCheckpoint = totalCheckpoints === 1;
+
+  return (
+    <div
+      id="checkpoint"
+      className="pgn__checkpoint p-4 shadow-lg"
+      aria-labelledby="pgn__checkpoint-title"
+      role="dialog"
+      style={{ visibility: checkpointVisible ? 'visible' : 'hidden', pointerEvents: checkpointVisible ? 'auto' : 'none' }}
+    >
+      {/* This text is not translated due to Paragon's lack of i18n support */}
```

Comment:
Jeff Witt's advice was to hardcode this into Paragon rather than relying on developer's to provide the sr-only text themselves.